pennylane/decomposition/__init__.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+.. currentmodule:: pennylane
+
+This module implements the infrastructure for PennyLane's new graph-based decomposition system.
+
+.. warning::
+
+    This module is experimental and is subject to change in the future.
+
+.. currentmodule:: pennylane.decomposition
+
+.. autosummary::
+    :toctree: api
+
+    ~register_resources
+    ~resource_rep
+    ~controlled_resource_rep
+    ~adjoint_resource_rep
+    ~add_decomps
+    ~list_decomps
+    ~has_decomp
+    ~Resources
+    ~DecompositionRule
+
+.. TODO::
+    Add section here explaining how to enable and disable the new decomposition system. [sc-83993]
+
+**Defining Decomposition Rules**
```

Comment:
Just a word of caution here

It's best to avoid having too many sources of truth. There are several proposed areas for having the same information: here, `register_resources`, and the compiling circuits page (see here: https://github.com/PennyLaneAI/pennylane/pull/7066), off the top of my head.

Reply:
I feel that the docstring for `register_resources` explains the entire process quite well already. If we want to avoid repeating ourselves in multiple sections, what should the module docstring contain?

Reply:
Nothing necessarily to act on, just a word of caution 😅. Just checking other modules, the module docstrings seem to repeat a lot of elements in the module functions themselves. I'm happy to leave this here, as the precedence is set and it's not a terrible place to also have this information. Just something to be aware of I guess 😁

Reply:
I restructured the module docstring, maybe you can take another look.